### DISCUSSION TITLE
Have some troubles understanding hw2

### DISCUSSION THREAD

**Discussion by KimSSung** [seominjoon/kaist-ai605#120] [Q&A] [answered]
While I read all of the discussions about hw2, I still have some questions...

1. What does **'train'** function do in **'BoW' and 'TFIDF' each** ?
(I understood as **add**: add document and its vector / **search**: find top-k indices among the 'added(existing)' documents)
2. Why should **TFIDF class** subclasses the **BagOfWords class** ? (For calculating TF ?)
3. Can I create the vocab(with SQUAD trainset) outside of BoW / TFIDF class? or should this be done inside the **train function**?

Thanks!

**Comment by seominjoon** [comment, marked as answer]
1. 'train' creates vocab for BoW and also IDF values for tfidf.
2. for OOP, sharing same method signatures.
3. It should be done in 'train' function.

**Reply by KimSSung** [reply]
Thanks professor!
